src/Validator/Constraints/HasBackgroundImageValidator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Validator\Constraints;
+
+use Setono\SyliusGiftCardPlugin\Model\GiftCardConfigurationImageInterface;
+use Setono\SyliusGiftCardPlugin\Model\GiftCardConfigurationInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+final class HasBackgroundImageValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof HasBackgroundImage) {
+            throw new UnexpectedTypeException($constraint, HasBackgroundImage::class);
+        }
+
+        if (!$value instanceof GiftCardConfigurationInterface) {
+            return;
+        }
+
+        $backgroundImage = $value->getBackgroundImage();
+        if (!$backgroundImage instanceof GiftCardConfigurationImageInterface || null === $backgroundImage->getPath()) {
```

Comment:
```suggestion
        if (null === $backgroundImage || null === $backgroundImage->getPath()) {
```

Reply:
Is there a specific reason for that ? Or just something like line length ?
I tend to prefer the `instanceof` because it then gives a clear idea of what we expect. (Side effect is autocompletion in IDE as well)